### PR TITLE
EVG-2334 correctly unprovision host on failure

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: hostinit/setup.go [hunk 1]
```diff
@@ -462,6 +462,15 @@ func (init *HostInit) expandScript(s string) (string, error) {
...
 }
 
+func unprovisionHost(h *host.Host) {
+	if h == nil {
+		return
+	}
+	if err := h.SetUnprovisioned(); err != nil {
+		grip.Errorf("unprovisioning host %s failed: %s", h.Id, err.Error())
+	}
+}
```
Comment (tychoish) [inline hostinit/setup.go]: this should be a method in the hosts packshr

File: hostinit/setup.go [hunk 2]
```diff
@@ -462,6 +462,15 @@ func (init *HostInit) expandScript(s string) (string, error) {
 	return script, err
 }
 
+func unprovisionHost(h *host.Host) {
+	if h == nil {
+		return
+	}
+	if err := h.SetUnprovisioned(); err != nil {
+		grip.Errorf("unprovisioning host %s failed: %s", h.Id, err.Error())
```
Comment (tychoish) [inline hostinit/setup.go]: I'd just do an errors wrap here.

Comment (john-m-liu) [reply hostinit/setup.go]: done

File: hostinit/setup.go (hunk 1 above)
Comment (john-m-liu) [reply hostinit/setup.go]: moved

File: model/host/host.go [hunk 3]
```diff
@@ -208,6 +208,10 @@ func (h *Host) SetUnprovisioned() error {
 			},
 		},
 	)
+	if err != nil {
+		grip.Error(errors.Wrapf(err, "error unprovisioning host %s", h.Id))
+	}
```
Comment (tychoish) [inline model/host/host.go]: don't return error; you can just log the errors.Wrap(err)

Comment (john-m-liu) [reply model/host/host.go]: done

**Comment by bsamek** [comment]
> ish - the monitor terminates unprovisioned hosts so it will get builder hosts. I think users need to manually terminate spawn hosts that are like this though

I think what used to happen is that the host would disappear and then users would receive an email that the spawn failed. It seems wrong to have an extra manual step, that users have to delete hosts that failed to spawn. I opened https://jira.mongodb.org/browse/EVG-2335 so that we can figure out the spawn host failure problem. On the other hand, I'm not especially wedded to this.

@tychoish Thoughts?

**Comment by bsamek** [comment]
> I'm not sure that fixing the underlying issue, notifying properly, and this patch are mutually exclusive, but I may be missing something.

I think that this work can be done either in the context of this PR or under a separate ticket. Up to @john-m-liu.

**Pull request merged**